packages/xod-arduino/platform/program.tpl.cpp

```diff
@@ -286,9 +285,15 @@ void runTransaction() {
           {{/eachDirtyablePin}}
 
           {{#if patch.raisesErrors}}
-            ErrorFlags previousErrorFlags = node_{{ id }}.errorFlags;
-            // give the node a chance to recover from it's own previous errors
-            node_{{ id }}.errorFlags = 0;
+            // TODO: a copy constructor to make this less ugly?
+            {{ns patch }}::NodeErrors previousErrors = { .flags=node_{{ id }}.errors.flags };
```

Comment:
Yes, please. But aren’t structs have a default copy-constructor which copies data over? I mean, to me it seems like an equivalent to trivial:

```cpp
{{ns patch }}::NodeErrors previousErrors = node_{{ id }}.errors;
```